### PR TITLE
Pretty xxh3

### PR DESCRIPTION
**Pull request by zwhitchcox**

### CODE REVIEW

File: lib/source-destination/source-destination.ts [hunk 1]
```diff
@@ -77,8 +42,41 @@ import {
...
+		super();
+		if (outEnc && typeof outEnc !== 'string' && !Buffer.isBuffer(outEnc)) {
+			outEnc = 'buffer';
+		}
+
+		this._outEnc = outEnc as string;
+
+		this._hash = new XXHash3(seed);
+
+		// Hack needed to support `readableObjectMode` in node v0.10 and to set
```
Comment (zvin) [inline lib/source-destination/source-destination.ts:59]: We don't care about node 0.10

File: lib/source-destination/source-destination.ts [hunk 2]
```diff
@@ -77,8 +42,41 @@ import {
...
+		this._hash.update(chunk);
+		callback();
+	}
+	_flush(callback: () => void) {
+		if (this._outEnc !== undefined) {
+			this.push(this._hash.digest(this._outEnc));
+		} else {
+			this.push(this._hash.digest());
+		}
+		callback();
```
Comment (zvin) [inline lib/source-destination/source-destination.ts:76]: Isn't this the same as `this.push(this._hash.digest(this._outEnc))` ?

File: lib/sparse-stream/shared.ts [hunk 3]
```diff
@@ -120,7 +108,7 @@ function verifyOrGenerateChecksum(
 		const checksum = hasher.digest('hex') as string;
 		if (generateChecksums) {
 			blocks.checksum = checksum;
-		} else if (verify && blocks.checksum !== checksum) {
+		} else if (verify && blocks.checksum?.toString() !== checksum.toString()) {
```
Comment (zvin) [inline lib/sparse-stream/shared.ts:111]: `blocks.checksum` is either `undefined` or a string, no need to call `.toString()` on it;
`checksum` is supposed to be a string to, at least according to the `as string above`

File: package.json [hunk 4]
```diff
@@ -68,11 +68,11 @@
     "outdent": "^0.7.0",
     "partitioninfo": "^6.0.2",
     "rwmutex": "^1.0.0",
+    "through": "^2.3.8",
```
Comment (zvin) [inline package.json:71]: Is this used anywhere ?

File: lib/sparse-stream/shared.ts (hunk 3 above)
Comment (zwhitchcox) [reply lib/sparse-stream/shared.ts:111]: @zvin Well, I think with the changes, is can either be a buffer or a string...I am not sure why this happened, but I will look into it.

File: lib/source-destination/source-destination.ts (hunk 2 above)
Comment (zwhitchcox) [reply lib/source-destination/source-destination.ts:76]: Yes, you're right

File: package.json (hunk 4 above)
Comment (zwhitchcox) [reply package.json]: Removed

**Comment by thundron** [comment]
cherry-picked in https://github.com/balena-io-modules/etcher-sdk/pull/213

**Pull request closed without merging**